src/formatDate.js

```diff
@@ -8,7 +8,36 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const year4 = 'YYYY';
+  const year2 = 'YY';
+
+  const separatedDate = date.split(fromFormat[3]);
+
+  const dateData = {};
+
+  for (let i = 0; i < separatedDate.length; i++) {
+    dateData[fromFormat[i]] = separatedDate[i];
+  }
+
+  if (!dateData[year2]) {
+    dateData[year2] = dateData[year4].slice(2, 4);
+  }
+
+  if (!dateData[year4]) {
+    if (dateData[year2] <= 20) {
+      dateData[year4] = `20${dateData[year2]}`;
+    } else {
+      dateData[year4] = `19${dateData[year2]}`;
+    }
+  }
+
+  const result = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    result.push(dateData[toFormat[i]]);
+  }
```

Comment:
The loop should include the last element of `toFormat` array, which is the new separator. You should not exclude it from the loop condition. Instead, handle the separator outside of the loop to avoid adding it to the `result` array.